src/main/java/com/minecolonies/coremod/colony/workorders/WorkOrderBuildDecoration.java

```diff
@@ -226,13 +227,38 @@ public void onCompleted(final IColony colony)
         {
             final int level = ((WorkOrderBuildBuilding) this).getUpgradeLevel();
             AdvancementUtils.TriggerAdvancementPlayersForColony(colony, player ->
-                    AdvancementTriggers.COMPLETE_BUILD_REQUEST.trigger(player, structureName, level));
+                                                                          AdvancementTriggers.COMPLETE_BUILD_REQUEST.trigger(player, structureName, level));
         }
         else
         {
+
+
             AdvancementUtils.TriggerAdvancementPlayersForColony(colony, player ->
-                    AdvancementTriggers.COMPLETE_BUILD_REQUEST.trigger(player, structureName, 0));
-            LanguageHandler.sendPlayersMessage(colony.getMessagePlayerEntities(),COM_MINECOLONIES_COREMOD_ENTITY_BUILDER_BUILDCOMPLETE,getStructureName());
+                                                                          AdvancementTriggers.COMPLETE_BUILD_REQUEST.trigger(player, structureName, 0));
+
+            String builder = "";
+            for (ICitizenData citizen : colony.getCitizenManager().getCitizens())
+            {
+                try
+                {
+                    if (citizen.getWorkBuilding().getPosition().equals(this.getClaimedBy()))
```

Comment:
why does this need a try and catch?

Reply:
Better change the onCompleted method to receive the citizenData of the citizen that completed the job.

Reply:
> why does this need a try and catch?

For some citizens it would cause a crash

Reply:
except a try & catch you could add a null check before like getWorkBuilding() != null &&

Reply:
Still working for the adjustment

Reply:
Testing to make sure I didn't break anything else.